aggregator/src/main/java/com/redhat/cloud/notifications/DailyEmailAggregationJob.java

```diff
@@ -111,10 +116,37 @@ public void processDailyEmail() {
         }
     }
 
-    List<AggregationCommand> processAggregateEmailsWithOrgPref(LocalDateTime endTime, CollectorRegistry registry) {
 
-        final List<AggregationCommand> pendingAggregationCommands =
-                emailAggregationResources.getApplicationsWithPendingAggregationAccordinfOrgPref(endTime);
+    /**
+     * Compute targeted schedule execution time local date time.
+     * User can schedule execution time every 15 min.
+     * Event if this job is executed at 05:02, we are looking for schedule executions of 05:00
+     *
+     * @return the local date time
+     */
+    protected LocalDateTime computeScheduleExecutionTime() {
+        final LocalDateTime currentTime = LocalDateTime.now(UTC).withSecond(0).withNano(0);
+
+        // correct Time
+        if (currentTime.getMinute() < 15) {
+            return currentTime.withMinute(0);
+        } else if (currentTime.getMinute() < 30) {
+            return currentTime.withMinute(15);
+        } else if (currentTime.getMinute() < 45) {
+            return currentTime.withMinute(30);
+        } else {
+            return currentTime.withMinute(45);
+        }
+    }
+
+    List<IAggregationCommand> processAggregateEmailsWithOrgPref(LocalDateTime endTime, CollectorRegistry registry) {
+
+        List<IAggregationCommand> pendingAggregationCommands;
+        if (aggregatorConfig.isAggregationBasedOnEventEnabled()) {
+            pendingAggregationCommands = emailAggregationResources.getApplicationsWithPendingAggregationAccordingOrgPref(endTime);
+        } else {
+            pendingAggregationCommands = emailAggregationResources.getApplicationsWithPendingAggregationAccordinfOrgPref(endTime);
```

Comment:
I stared at this for a full minute before spotting the tiny difference between the two method names... 🤯

Was that intended?

Reply:
Yes the old one had a typo, so I fixed it with the new one.